### PR TITLE
Including documentation about the STH component in the README.md file

### PR DESCRIPTION
**Pull request by gtorodelvalle**
- Implements https://github.com/telefonicaid/IoT-STH/issues/4

- 100% unit tests passed:
  - To run the tests: make test

- Review assigned to @frbattid

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,235 @@
...
+* [Dependencies](#section2)
+* [Installation](#section3)
+* [Inserting data (single events and aggregated data) into the database](#section4)
+* [Querying data from the database](#section5)
+* [Running the STH server](#section6)
+* [Contact](#section7)
+
+##<a id="section1"></a> Introduction
+The STH component is a FIWARE component in charge of providing aggregated time series information about the evolution of
+entity attribute values in time.
```
Comment (frbattid) [inline README.md]: Some references to Orion Context Broker and the NGSI model would be necessary.

File: README.md [hunk 2]
```diff
@@ -1 +1,235 @@
...
+
+The aggregated time series information is stored in a MongoDB instance. This information can be generated by 2 main
+means:
+
+1. The STH component can directly subscribe to the Context Broker to receive notifications when the entity attribute
+values change, calculating the aggregated time series information and storing it in the MongoDB instance.
+This option is called the minimalist option.
+2. A new sink can be enabled in the Cygnus component to calculate and to update the aggregated time series information
```
Comment (frbattid) [inline README.md]: Some reference (link) to Cygnus would be necessary.

File: README.md [hunk 3]
```diff
@@ -1 +1,235 @@
...
+as the entity attribute values change over time. This option is the formal one.
```
Comment (frbattid) [inline README.md]: Some way, I would say that using Cygnus a set of common features to this backend and others is available. This is in order a newbie can decide among the minimalist and the formal options.

File: README.md [hunk 4]
```diff
@@ -1 +1,235 @@
...
+
+Regarding the aggregated time series information provided by the STH component, there are 4 main concepts which are
+important to know about:
+
+* <b>Range</b>: The period of time about which the aggregated time series information is provided. Possible valid
+ranges values are: year, month, day, hour, minute.
+* <b>Resolution</b> or <b>aggregation period</b>: The time period by which the aggregated time series information is grouped.
+Possible valid resolution values are: month, day, hour, minute and second. For the time being, we only consider the
```
Comment (frbattid) [inline README.md]: Unnecessary line feed/carriage return?

File: README.md [hunk 5]
```diff
@@ -1 +1,235 @@
...
+This option is called the minimalist option.
+2. A new sink can be enabled in the Cygnus component to calculate and to update the aggregated time series information
+as the entity attribute values change over time. This option is the formal one.
+
+Regarding the aggregated time series information provided by the STH component, there are 4 main concepts which are
+important to know about:
+
+* <b>Range</b>: The period of time about which the aggregated time series information is provided. Possible valid
+ranges values are: year, month, day, hour, minute.
+* <b>Resolution</b> or <b>aggregation period</b>: The time period by which the aggregated time series information is grouped.
```
Comment (frbattid) [inline README.md]: Unnecessary line feed/carriage return?

File: README.md [hunk 6]
```diff
@@ -1 +1,235 @@
...
+Possible valid resolution values are: month, day, hour, minute and second. For the time being, we only consider the
+following range-resolution pairs: year-month, month-day, day-hour, hour-minute and minute-second.
+* <b>Origin</b>: For certain range-resolution pair, it is the origin of time for which the aggregated time series
+information applies. For example, for a pair hour-minute, a valid origin value could be: ``2015-03-01T13:00:00.000Z```,
```
Comment (frbattid) [inline README.md]: A simple quote character is missing before `2015-03-...` or an unnecessary quote character appears after `...00.000Z`.

File: README.md [hunk 7]
```diff
@@ -1 +1,235 @@
...
+meaning the 13th hour of March, the 3rd, 2015. The origin is stored using UTC time to avoid locale issues.
+* <b>Offset</b>: For certain range-resolution pair, it is the offset from the origin for which the aggregated time series
+information applies. For example, for a pair hour-minute and an origin ```2015-03-01T13:00:00.000Z```, an offset of 10
+refers to the 10th minute of the concrete hour pointed by the origin. In this example, there would be a maximum of 60
+offsets from 0 to 59 corresponding to each one of the 60 minutes within the concrete hour.
+* <b>Samples</b>: For a quadruple range-resolution-origin-offset, it is the number of samples, values, events or notifications available.
+
+The STH component exposes an HTTP ReST API to let external clients query this aggregated time series information. A
```
Comment (frbattid) [inline README.md]: ReST instead of REST or Rest

File: README.md [hunk 8]
```diff
@@ -1 +1,235 @@
...
+
+<pre>http://localhost:8666/STH/v1/contextEntities/type/&lt;quantity&gt;/id/&lt;entityId&gt;/attributes/&lt;attributeId&gt;?aggrMethod=sum&aggrPeriod=second&dateFrom=2015-02-22T00:00:00&dateTo=2015-02-22T23:00:00</pre>
+
+The entries between "<" and ">" in the URL path depend on the concrete case (type of data, entity and attribute) being queried.
+
+The requests can make use the following query parameters:
+
+* <b>aggrMethod</b>: The aggregation method. The STH component supports the following aggregation methods: max (maximum
+value), min (minimum value), sum (sum of all the samples) and sum2 (sum of the square value of all the samples). Combining
+the information provided by these aggregated methods with the number of samples, it is possible to calculate probabilistic
```
Comment (frbattid) [inline README.md]: Unnecessary line feed/carriage return?

File: README.md [hunk 9]
```diff
@@ -1 +1,235 @@
-# IoT-STH
+#<a id="section0"></a> IoT-STH
+
+* [Introduction] (#section1)
+* [Dependencies](#section2)
+* [Installation](#section3)
+* [Inserting data (single events and aggregated data) into the database](#section4)
+* [Querying data from the database](#section5)
+* [Running the STH server](#section6)
```
Comment (frbattid) [inline README.md]: I would describe how to start the server inmmediatelly after the installation step.

File: README.md [hunk 10]
```diff
@@ -1 +1,235 @@
...
+<pre> git clone https://github.com/telefonicaid/IoT-STH.git </pre>
+2. Get into the directory where the STH repository has been cloned:
+<pre> cd IoT-STH/ </pre>
+3. Install the Node.js modules and dependencies:
+<pre> npm install </pre>
+The STH component server is ready to be started.
+
+[Top](#section0)
+
+##<a id="section4"></a> Inserting data (single events and aggregated data) into the database
```
Comment (frbattid) [inline README.md]: I would say something like "Inserting data (random single events and aggregated data) into the database".

File: README.md (hunk 1 above)
Comment (gtorodelvalle) [inline README.md]: Fixed in https://github.com/telefonicaid/IoT-STH/commit/0ec29adce002b71a13c80a81df031be12902ecb6

File: README.md (hunk 2 above)
Comment (gtorodelvalle) [inline README.md]: Fixed in https://github.com/telefonicaid/IoT-STH/commit/4156f6ae61af82c5d813d9dcbdda6f65d3fc6b7b

File: README.md (hunk 3 above)
Comment (gtorodelvalle) [inline README.md]: Fixed in https://github.com/telefonicaid/IoT-STH/commit/65bdea76e041e270660588e4979d6b49ac01947d

File: README.md (hunk 5 above)
Comment (gtorodelvalle) [inline README.md]: Yeah, I included it for an easier reading in an IDE :) Do you prefer avoiding this carriage returns? :)

File: README.md (hunk 6 above)
Comment (gtorodelvalle) [inline README.md]: Fixed in https://github.com/telefonicaid/IoT-STH/commit/8e06af67b4a347a8614e438a3b3ab97e6a71333b

File: README.md (hunk 7 above)
Comment (gtorodelvalle) [inline README.md]: Fixed in https://github.com/telefonicaid/IoT-STH/commit/553c8aa81a09b7fbd880885e8db989946e1041e6

File: README.md (hunk 8 above)
Comment (gtorodelvalle) [inline README.md]: Yeah, I included it for an easier reading in an IDE :) Do you prefer avoiding this carriage returns? :)

File: README.md (hunk 10 above)
Comment (gtorodelvalle) [inline README.md]: Fixed in https://github.com/telefonicaid/IoT-STH/commit/422dde2e4df20d47b972c2dc549664b1254c9cd5

File: README.md (hunk 5 above)
Comment (frbattid) [inline README.md]: It is not important, NTC :)

File: README.md (hunk 8 above)
Comment (frbattid) [inline README.md]: It is not important, NTC :)

**Comment by frbattid** [comment]
LGTM, good job!

**Pull request merged**